src/gui/mainwindow.cpp

```diff
@@ -329,12 +331,32 @@ void MainWindow::restoreWindowGeometry()
 	qRegisterMetaTypeStreamOperators<QList<int>>("QList<int>");
 #endif
 
-	QSettings settings("nvim-qt", "window-geometry");
+	const QSettings settings("nvim-qt", "window-geometry");
 	if (!settings.value("restore_window_geometry", true).toBool()) {
 		return;
 	}
-	restoreGeometry(settings.value("window_geometry").toByteArray());
-	restoreState(settings.value("window_state").toByteArray());
+	const QVariant geometry{ settings.value("window_geometry") };
+	if (geometry.isValid()) {
+		restoreGeometry(geometry.toByteArray());
+	}
+	else {
+		const QPoint local_position{ width() / 2, 0 };
+		const QPoint global_position = mapToGlobal(local_position);
+		const QScreen* screen{ qApp->screenAt(global_position) };
+		if (screen) {
+			const QRect geometry{ screen->availableGeometry() };
+			if (screen->orientation() == Qt::LandscapeOrientation) {
+				resize(geometry.width() / 2, geometry.height());
+			}
+			else {
+				resize(geometry.width(), geometry.height() / 2);
+			}
+		}
```

Comment:
Last comment, I promise :)

Let's keep this block as a function, so everything is modular and easy to read.